### ISSUE TITLE
ndmap does not pickle properly

### ISSUE THREAD

**Issue by mhasself** [simonsobs/pixell#237]
* pixell version: 0.20.4
* Python version: 3.10.12
* Operating System:  Linux

### Description

I was trying to pickle a small ndmap.  On load, it reconstitutes as an ndmap object, but without the `.wcs` attribute set.

It might be difficult to get ndmaps to pickle ... not a big deal.  But it would be good to raise an exception, for now, indicating it's not supported.

### What I Did

Test script:
```
from pixell import enmap
import pickle

g = enmap.fullsky_geometry(res=10 * enmap.utils.degree)
m0 = enmap.zeros(*g)

pickle.dump({'map': m0},
            open('test.pik', 'wb'))

m1 = pickle.load(open('test.pik', 'rb'))['map']

print(type(m0), hasattr(m0, 'wcs'))
print(type(m1), hasattr(m1, 'wcs'))
```

Has output:
```
<class 'pixell.enmap.ndmap'> True
<class 'pixell.enmap.ndmap'> False
```

**Comment by msyriac** [comment]
I don't have a problem with proper pickling being implemented (we should certainly raise an error or warning until then), but is there ever a case where pickling is encouraged over writing to FITS or HDF5?

**Comment by mhasself** [comment]
I don't think we need to encourage pickling; it's just a convenience.  In my use case, I was using it to store some calculated state (an object with several properties, including a few low-res sky maps), so that I could debug the subsequent processing without suffering the per-iteration cost of recalculating my initial state.

**Comment by msyriac** [comment]
Got it, thanks.
